helix-term/src/ui/editor.rs

```diff
@@ -1182,12 +1183,19 @@ impl Component for EditorView {
                         // how we entered insert mode is important, and we should track that so
                         // we can repeat the side effect.
 
-                        self.last_insert.0 =
-                            match self.keymaps.get_mut(&mode).unwrap().get(key).kind {
-                                KeymapResultKind::Matched(command) => command,
-                                // FIXME: insert mode can only be entered through single KeyCodes
-                                _ => unimplemented!(),
-                            };
+                        self.last_insert.0 = match self
+                            .keymaps
+                            .load()
+                            .clone()
+                            .get_mut(&mode)
```

Comment:
I'm not clear on why this was doing `get_mut()`, you could probably avoid the `.clone()` by using `.get()`

Reply:
From a glance, it looks like it was done because of the `unwrap()` after it